runtime/vm/jvminit.c

```diff
@@ -2844,6 +2844,23 @@ modifyDllLoadTable(J9JavaVM * vm, J9Pool* loadTable, J9VMInitArgs* j9vm_args)
 					return JNI_ERR;
 				}
 			}
+
+			/*
+			On Linux on Z it needs to open the libj9zlib to load libj9jit as it is used for AOT method
+			data compression which is currently only enabled on Z platform. To make sure when JVM loads 
+			libj9jit, libj9zlib is already loaded, we are openning it.
+			For more details see OpenJ9 PR#8561
+			*/
+			#if (defined(S390) && defined(LINUX))
+				char zlibDll[EsMaxPath];
+				char *zlibDllDir = zlibDll;
+				strcpy(zlibDllDir, vm->j9libvmDirectory);
+				strncat(zlibDllDir, DIR_SEPARATOR_STR, sizeof(DIR_SEPARATOR_STR)-1);
+				strncat(zlibDllDir, J9_ZIP_DLL_NAME, sizeof(J9_ZIP_DLL_NAME)-1);
+				if (0 != j9sl_open_shared_library(zlibDllDir, &(entry->descriptor), openFlags))
+ 					j9tty_printf(PORTLIB, "Error: Failed to open zlib DLL %s (%s)\n", zlibDllDir, j9error_last_error_message());
```

Comment:
If the preload fails, shouldn't this return an error and prevent the VM from starting?